internal/monitor/metric.go

```diff
@@ -0,0 +1,55 @@
+package monitor
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/razorpay/trino-gateway/internal/boot"
+)
+
+type Metrics struct {
+	env                string
+	executionsTotal    *prometheus.CounterVec
+	executionlastRunAt *prometheus.GaugeVec
+	executionDurations *prometheus.HistogramVec
+	backendLoad        *prometheus.GaugeVec
+}
+
+var metrics *Metrics
+
+func initMetrics() {
+	metrics = &Metrics{
+		env: boot.Config.App.Env,
+	}
+	metrics.executionsTotal = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "trino_gateway_monitor_executions_total",
+			Help: "Number of executions triggered for monitor task.",
+		},
+		[]string{"env"},
+	)
+
+	metrics.executionlastRunAt = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "trino_gateway_monitor_execution_last_run_at",
+			Help: "Monitor task last run epoch ts.",
+		},
+		[]string{"env"},
+	)
+
+	metrics.executionDurations = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "trino_gateway_monitor_execution_durations_s_histogram",
```

Comment:
nit: suggesting rename to `trino_gateway_monitor_execution _duration_secs_histograms` or `trino_gateway_monitor_execution_seconds_histogram`

Reply:
renaming to `trino_gateway_monitor_execution_seconds_histogram `